### ISSUE TITLE
Sustained voices are held after "Reset all controllers"

### ISSUE THREAD

**Issue by rianhunter** [FluidSynth/fluidsynth#1049]
### FluidSynth version

```
$ fluidsynth --version
FluidSynth runtime version 2.1.7
Copyright (C) 2000-2021 Peter Hanappe and others.
Distributed under the LGPL license.
SoundFont(R) is a registered trademark of E-mu Systems, Inc.

FluidSynth executable version 2.1.7
Sample type=double
```

### Describe the bug

According to the GM recommendations in the MIDI spec, "Reset All Controllers" should reset all pedals (#<span/>64-#<span/>67) to 0. This includes sustain. If a note is currently held in sustain, it is not turned off after "Reset all Controllers" like it would be if the sustain pedal was turned off normally.

### Expected behavior

I would expect that all notes held under sustain would turn off when the sustain pedal is set to 0.

### Steps to reproduce

1. Turn a note on
2. Set sustain CC (#<span/>64)
3. Send note off (note should still be playing)
4. Send "reset all controllers" CC (#<span/>121)
5. Note is still playing

**Comment by derselbst** [comment]
Nice finding, thanks! I've just created a [regression test](https://github.com/FluidSynth/fluidsynth/blob/reset-cc/test/test_synth_reset_cc.c) for this issue. It's pretty obvious when looking at the code, because we're missing to call `fluid_synth_damp_voices_by_sustain_LOCAL` in case `ALL_CRTL_OFF` was called before. Same is true for Sostenuto.

There are several ways to address this, I'll need to think about it. I'll look into the PORTAMENTO issue afterwards.

**Comment by rianhunter** [comment]
No problem. To be honest I noticed this issue but I'm not sure what the GM spec expects. Turning notes off after "reset all controllers" seems like the only logical conclusion but it still feels weird IMO. Ideally I think it would be better to not touch pedal state on "reset all controllers" at all. What do you think about that?

**Comment by rianhunter** [comment]
After some more thought I now think the correct thing to do is release the sustained notes on "reset all controllers." From what I can tell "reset all controllers" is primarily intended to reset performance controllers so that after a sequencer is done playing a MIDI sequence it sends "reset all controllers" so that if the user starts playing the keyboard it isn't stuck in a pitch bend or similar. In this way it's similar "All notes off." 

Given this, it's probably a good idea to not reset the channel specific generators. You wouldn't want the current sound synthesis to sound different just because the sequencer is stopped. At least that's my interpretation but that is another issue entirely.

**Comment by derselbst** [comment]
Fixed and will be part of 2.2.6. Thanks again!
